src/main.rs

```diff
@@ -62,18 +62,12 @@ fn data_decode(str: &str) -> Result<(usize, Vec<u8>)> {
     Ok((length, bytes))
 }
 
-fn parse(path: &Path) -> Result<Recording> {
-    let f = std::fs::File::open(path)?;
-    let lines = std::io::BufReader::new(f).lines();
+fn parse(lines: &[String], mut stderr: impl std::io::Write) -> Result<Recording> {
```

Comment:
It was already the case previously, but you likely want to lazy parse the incoming file. If the file is too big because it was recorded for too long you should just parse the lines as they com in, not load the entire file in memory first.

So IMO your `parse()` should take a single line and return a boolean to say continue/stop, and you should call this in a `.map()` call.

Reply:
done now, took a while until I grasped it but the key was to use `impl Iterator` as function argument